### ISSUE TITLE
Babel doesn't watch for file change

### ISSUE THREAD

**Issue by ifndefdeadmau5** [babel/example-node-server#38]
For now I just restart manually after making changes because babel doesn't automatically transpile original code when changes are made. Should I use another library for watching purpose?

**Comment by bryandbor** [comment]
I agree with this. I can see that `nodemon` notices a change and restarts; however, it doesn't transpile the code and rerun the new code. That kind of defeats the purpose of `nodemon` watching for changes, doesn't it?

**Comment by ifndefdeadmau5** [comment]
@bryandbor Thank you for help! It works nicely. but it seems little bit slow because it builds whole code for every changes. Do you think any better solution is exist?

**Comment by SeyTo** [comment]
@ifndefdeadmau5 Maybe this could work.
`"serve": "babel -w lib -d dist & nodemon dist --delay 800ms "`
babel should just watch changes and convert to `dist`. Then `nodemon` picks up the new changes in the `dist` directory. The delay is be needed as nodemon will go crazy and restart everytime babel creates new files.

**Comment by Miikis** [comment]
Ran into a similar problem. Here's my working implementation:

```js
"main": "dist/server",
 "scripts": {
        "clean": "rimraf dist",
        "nodemon:watch": "nodemon -e js,pug --ignore tests/ -r dotenv/config .",
        "build": "npm run clean && babel src -d dist --copy-files",
        "build:watch": "babel src --watch -d dist --copy-files --source-maps inline",
        "start": "npm run build && NODE_ENV=production node -r dotenv/config .",
        "start:watch": "run-p build:watch nodemon:watch",
        "test": "NODE_ENV=test jest tests/unit-tests/*.test.js --notify",
        "test:watch": "NODE_ENV=test jest tests/unit-tests/*.test.js --watchAll",
        "watch": "npm run build:prod && run-p build:dev start:dev"
 },
"devDependencies": {
        "@babel/cli": "^7.1.2",
        "@babel/core": "^7.1.2",
        "@babel/preset-env": "^7.1.0",
        "@babel/register": "^7.0.0",
        "babel-core": "^7.0.0-bridge.0",
        "babel-jest": "^23.6.0",
        "jest": "^23.6.0",
        "nodemon": "^1.18.7",
        "npm-run-all": "^4.1.5",
        "rimraf": "^2.6.2"
}
```

**Don't forget to install the dev-deps.**

**Comment by SmolinPavel** [comment]
> @bryandbor Thank you for help! It works nicely. but it seems little bit slow because it builds whole code for every changes. Do you think any better solution exist?

@ifndefdeadmau5 I think the solution you want is something like 

`"start": "nodemon --exec babel-node lib/index.js",`

So what happens here - we just say nodemon to look for the initial file which is `lib/index.js`

**Comment by SaulSilver** [comment]
@SmolinPavel Thank you for the solution. Just to note that Babel do not recommend using `babel-node` for production as they state here: https://babeljs.io/docs/en/babel-node#not-meant-for-production-use
> Not meant for production use
You should not be using babel-node in production. It is unnecessarily heavy, with high memory usage due to the cache being stored in memory. You will also always experience a startup performance penalty as the entire app needs to be compiled on the fly.

**Comment by SmolinPavel** [comment]
@SaulSilver Oh yeah, good shout! 👍 Thanks

**Comment by ser14joker** [comment]
You can set up your scripts like this:
```
"scripts": {
    "dev": "nodemon lib/index.js",
    "watch": "babel src --watch --out-dir lib"
  }
```
and run them in separate terminals; **yarn watch** or **npm run watch** will watch changes on **src** folder and **yarn dev** or **npm run dev** will run the compiled file with **nodemon**. 🙏
